Unknown Error When Launching Server /w Port Forwarding On

I'm having this weird error when trying to get a server up and running
## Port Forwarding Off

```
####@dpServer19:~/Gameserver/_openra/_ra_casual$ ./run
[2016-10-19T10:36:09] Starting dedicated server for mod: ra
[2016-10-19T10:36:14] Master server communication established.
[2016-10-19T10:36:14] Warning: Server ports are not forwarded.
[2016-10-19T10:36:14] Game has not been advertised online.
[2016-10-19T10:36:50] User has joined the game.
[2016-10-19T10:37:06] User changed the map to ExampleMap1.
[2016-10-19T10:37:06] Singleplayer games have been disabled on this server.
[2016-10-19T10:37:19] User changed the map to ExampleMap2.
[2016-10-19T10:37:19] Singleplayer games have been disabled on this server.
[2016-10-19T10:37:24] User has disconnected.
^C
```
## Port Forwarding On

```
####@dpServer19:~/Gameserver/_openra/_ra_casual$ ./run
[2016-10-19T10:38:35] Starting dedicated server for mod: ra

Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Network.UPnP.ForwardPort (Int32 lifetime) <0x41e45970 + 0x000ac> in <filename unknown>:0 
  at OpenRA.Server.Server..ctor (System.Net.IPEndPoint endpoint, OpenRA.ServerSettings settings, OpenRA.ModData modData, Boolean dedicated) <0x41e1dae0 + 0x00503> in <filename unknown>:0 
  at OpenRA.Server.Program.Main (System.String[] args) <0x41d94830 + 0x00283> in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Network.UPnP.ForwardPort (Int32 lifetime) <0x41e45970 + 0x000ac> in <filename unknown>:0 
  at OpenRA.Server.Server..ctor (System.Net.IPEndPoint endpoint, OpenRA.ServerSettings settings, OpenRA.ModData modData, Boolean dedicated) <0x41e1dae0 + 0x00503> in <filename unknown>:0 
  at OpenRA.Server.Program.Main (System.String[] args) <0x41d94830 + 0x00283> in <filename unknown>:0 
^C

```

Port forwarding has been configured and is working however I am still getting this error.
Server was running on Ubuntu 16.04.

On an unrelated note, is there any setting to set the default map?

--- a/OpenRA.Game/Server/Server.cs
+++ b/OpenRA.Game/Server/Server.cs
@@ -138,7 +138,8 @@ namespace OpenRA.Server
 
 			randomSeed = (int)DateTime.Now.ToBinary();
 
-			if (Settings.AllowPortForward)
+			// UPnP is only supported for servers created by the game client.
+			if (!dedicated && Settings.AllowPortForward)
 				UPnP.ForwardPort(Settings.ListenPort, Settings.ExternalPort).Wait();
 
 			foreach (var trait in modData.Manifest.ServerTraits)
@@ -209,7 +210,7 @@ namespace OpenRA.Server
 					if (State == ServerState.ShuttingDown)
 					{
 						EndGame();
-						if (Settings.AllowPortForward)
+						if (!dedicated && Settings.AllowPortForward)
 							UPnP.RemovePortForward().Wait();
 						break;
 					}

--- a/launch-dedicated.cmd
+++ b/launch-dedicated.cmd
@@ -7,12 +7,11 @@ set Mod=ra
 set ListenPort=1234
 set ExternalPort=1234
 set AdvertiseOnline=True
-set AllowPortForward=False
 set EnableSingleplayer=False
 set Password=""
 
 :loop
 
-OpenRA.Server.exe Game.Mod=%Mod% Server.Name=%Name% Server.ListenPort=%ListenPort% Server.ExternalPort=%ExternalPort% Server.AdvertiseOnline=%AdvertiseOnline% Server.AllowPortForward=%AllowPortForward% Server.EnableSingleplayer=%EnableSingleplayer% Server.Password=%Password%
+OpenRA.Server.exe Game.Mod=%Mod% Server.Name=%Name% Server.ListenPort=%ListenPort% Server.ExternalPort=%ExternalPort% Server.AdvertiseOnline=%AdvertiseOnline% Server.EnableSingleplayer=%EnableSingleplayer% Server.Password=%Password%
 
 goto loop

--- a/launch-dedicated.sh
+++ b/launch-dedicated.sh
@@ -11,13 +11,12 @@ Mod="${Mod:-"ra"}"
 ListenPort="${ListenPort:-"1234"}"
 ExternalPort="${ExternalPort:-"1234"}"
 AdvertiseOnline="${AdvertiseOnline:-"True"}"
-AllowPortForward="${AllowPortForward:-"False"}"
 EnableSingleplayer="${EnableSingleplayer:-"False"}"
 Password="${Password:-""}"
 
 while true; do
      mono --debug OpenRA.Server.exe Game.Mod=$Mod \
      Server.Name="$Name" Server.ListenPort=$ListenPort Server.ExternalPort=$ExternalPort \
-     Server.AdvertiseOnline=$AdvertiseOnline Server.AllowPortForward=$AllowPortForward \
+     Server.AdvertiseOnline=$AdvertiseOnline \
      Server.EnableSingleplayer=$EnableSingleplayer Server.Password=$Password
 done

